linera-service/src/chain_listener.rs

```diff
@@ -123,30 +123,29 @@ where
         let client = {
             let mut map_guard = clients.map_lock().await;
             let context_guard = context.lock().await;
-            map_guard
-                .entry(chain_id)
-                .or_insert_with(|| {
-                    let client = context_guard.make_chain_client(storage.clone(), chain_id);
-                    Arc::new(Mutex::new(client))
-                })
-                .clone()
+            let btree_map::Entry::Vacant(entry) = map_guard.entry(chain_id) else {
+                return Ok(());
+            };
+            let client = context_guard.make_chain_client(storage.clone(), chain_id);
+            let client = Arc::new(Mutex::new(client));
+            entry.insert(client.clone());
+            client
         };
         let mut stream = ChainClient::listen(client.clone()).await?;
+        tokio::spawn(async move { while stream.next().await.is_some() {} });
```

Comment:
Maybe `tokio::spawn(stream.for_each(|_| future::ready(())));` ?

Reply:
Since we are not consuming the notification stream from `ChainClient::listen` any more, I believe we should take simplify the code and resolve #940.

Also, the same fix applies to the `Watch` command in `linera.rs`.

Reply:
Minor digression: I think another alternative could be

```suggestion
        tokio::spawn(stream.forward(sink::drain()));
```

Reply:
I like that, although it needs a `.map(Ok)` to make it implement `TryStream<Error = Infallible>`. :upside_down_face: